### ISSUE TITLE
Handle buildSteps metadata for builders registered from extensions

### ISSUE THREAD

**Issue by lingyuncai** [intel/webinizer#18]

**Comment by lingyuncai** [comment]
A possible quick solution is #21.

Currently, the metadata conversion between `builders` from `config.json` and `buildSteps` from `package.json` is based on the `builder __type__ <-> command` mapping. This is direct but has some limitations as below.

```cjs
// metadata of a buildStep
{
  "command": "emcmake cmake",
  "args": "",
  "cwd": "${projectRoot}"
}
```

```cjs
// corresponding builder field from config.json
{
  "__type__": "CMakeBuilder",
  "id": 0,
  "desc": "CMake",
  "command": "emcmake cmake",
  "args": "",
  "rootBuildFilePath": "${projectRoot}"
}
```

- The `builder __type__ <-> command` mapping should be unique, otherwise the conversion could be wrong. So it has limitation on such scenario: if user creates a customized builder in the extension that has the same command used for build as other builders registered, it would fail to convert such a buildStep defined in `package.json` to a registered builder.

- When converting a buildStep from `package.json` to a registered builder, if the command is not available from the mapping, it will be default to a `NativeBuilder` as it runs any user defined native commands (such as ls, cp, mv, etc..) so the command used is not fixed, which will also introduce a wrong conversion even without a warning.

- When a package in the dependency tree that uses builders provided by 3rd-party extensions (not available in our codebase), we have to install them manually. If the dependency tree is deep, this is a cumbersome work.

**Comment by lingyuncai** [comment]
Merged the quick solution in #21 to fix the issue and close. Please re-open if better solutions are available.
